app/src/main/java/org/simple/clinic/scanid/PatientPrefillInfo.kt

```diff
@@ -0,0 +1,32 @@
+package org.simple.clinic.scanid
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
+data class PatientPrefillInfo(
```

Comment:
So in the end `PatientPrefillInfo` class is only used as a data class for `PatientEntryScreen` to prefill.

Reply:
We can still add the different JSON object class for other QR codes if needed, right? I'm not clear about why do we need it? Currently `PatientPrefillInfo` class can also be used as a data class to prefill, right?

Reply:
So for first question, we would create different JSON object classes for parsing and we would need to pass those to patient entry right? Because currently with prefill info class we are doing that. We are parsing directly with prefill info class and passing that to patient entry.

Instead what we should be doing is, we need to parse it with a different JSON object class, and then map it to `PatientPrefillInfo` class. That way even if we add new JSON object class in future, we would still end up mapping it to `PatientPrefillInfo` and sending it to the `PatientEntryScreen`. Does that make sense?

Reply:
For the second question, yes, we are using `PatientPrefillInfo` class to prefill. But at the same time, we are using it to parse as well. So currently the variables/implementation is tied to NHID. So in future if we want to support other countries JSON we would end up creating a different prefill info class and using that for parsing different country.

Instead of doing all that we will use a different class for parsing and map the results to `PatientPrefillInfo`.

Reply:
For a better example, consider our current data. We have a payload object and then we have our Room entities and then we have data classes that are specific to our UI rendering needs, right? So, we will be following a similar pattern, where we don't expose the JSON data directly to the entry screen and instead just provide `PatientPrefillInfo` class.

Reply:
The example made sense, understood.